src/components/task-step/ends.cjsx

```diff
@@ -26,22 +36,96 @@ PracticeEnd = React.createClass
     </div>
 
 HomeworkEnd = React.createClass
-  render: ->
+  displayName: 'HomeworkEnd'
+
+  propTypes:
+    taskId: React.PropTypes.any.isRequired
+
+  _addListener: ->
+    TaskStepStore.addChangeListener(@_update)
+
+  _removeListener: ->
+    TaskStepStore.removeChangeListener(@_update)
+
+  componentWillMount:   -> @_addListener()
+  componentWillUnmount: -> @_removeListener()
+  componentWillUpdate: -> @_removeListener()
+  componentDidUpdate:  -> @_addListener()
+  # override focus from within task steps to focus on first to do question
+  componentDidMount: -> document.querySelector('.task-step textarea')?.focus()
+
+  _update: -> @setState({})
+
+  goToStep: () ->
+  onNextStep: () ->
+
+  renderReviewSteps: (steps, label, type) ->
+    stepsList = _.map steps, (step) =>
+      <TaskStep
+        id={step.id}
+        goToStep={@goToStep}
+        onNextStep={@onNextStep}
+      />
+
+    stepsReview =
+      <div className="task task-review-#{type}">
+        {label}
+        {stepsList}
+      </div>
+
+  renderAfterDue: (taskId) ->
+    completedSteps = TaskStore.getCompletedSteps taskId
```

Comment:
The maps aren't 1-liners. but yeah, we should have simpler code style rules. Maybe meet after the demo and discuss?